lib/Driver.js

```diff
@@ -83,80 +84,81 @@ module.exports = class Driver {
    * @returns {Promise<DidDocument>}
    */
   async generate({keyType = 'Ed25519VerificationKey2018'} = {}) {
-    const publicKey = await LDKeyPair.generate({type: keyType});
-    return this.keyToDidDoc(publicKey);
+    const keyPair = await cryptoLd.generate({type: keyType});
+    return this.keyPairToDidDocument({keyPair});
   }
 
   /**
    * Converts an Ed25519KeyPair object to a `did:key` method DID Document.
    *
-   * @param {Ed25519KeyPair} edKey
-   * @returns {DidDocument}
+   * @param {Ed25519KeyPair} keyPair - The key pair to use to generate the
+   *   DID Document.
+   * @returns {object} - An object containing the `didDocument` and `keyPairs`.
    */
-  async keyToDidDoc(edKey) {
-    const did = `did:key:${edKey.fingerprint()}`;
-    const keyId = `${did}#${edKey.fingerprint()}`;
-    edKey.controller = did;
+  async keyPairToDidDocument({keyPair} = {}) {
```

Comment:
Wasn't this more correct before ... that you only needed a public key to generate the DID Doc? Not the full key pair? Again, not sure how much longer we want to try and deal with this, but this is confusing to me and I expect it to generally be confusing to other developers. I suspect we don't have the primitives/abstractions quite right.